### PR TITLE
add toString implementation for UpdateRequest.

### LINKED ISSUE #27986: UpdateRequest need toString()
**Describe the feature**:
When using the Java API to do the update/index/delete, I log the request for debug. But the `UpdateRequest` not implemented the `toString()` method, which causes some inconvenience.

### PR DESCRIPTION
**Pull request by zzt93**
add toString simple implementation and test to resolve [issue 27986](https://github.com/elastic/elasticsearch/issues/27986)

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by zzt93** [comment]
Gradle check complains about the use of `java.lang.System#out` for test `toString`, so any suggestions?

**Comment by tvernum** [comment]
> Gradle check complains about the use of java.lang.System#out

I don't follow why you're trying to use `System.out` in that test. 
If you want to test `toString` you should just call that method, there's no reason to print it to standard output.
Instead, you should call `toString()` and check that the resulting `String` contains the text that you expect to see.

**Comment by zzt93** [comment]
Thanks, I see. Updated now.

**Comment by tvernum** [comment]
@elasticmachine ok to test

File: core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java [hunk 1]
```diff
@@ -893,4 +894,31 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
         return builder;
     }
+
+    @Override
+    public String toString() {
+        String res = "update {[" + index + "][" + type + "][" + id + "]";
```
Comment (DaveCTurner) [inline core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java]: There's lots of appending going on - could you use a `StringBuilder` here?

File: core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java [hunk 2]
```diff
@@ -893,4 +894,31 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
...
+        if (doc != null) {
+            res += (", doc[" + doc + "]");
+        }
+        if (script != null) {
+            res += (", script[" + script + "]");
+        }
+        if (upsertRequest != null) {
+            res += (", upsert[" + upsertRequest + "]");
+        }
+        if (scriptedUpsert) {
```
Comment (DaveCTurner) [inline core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java]: This is `boolean` - I think it'd be better to include it in the output whether it's `true` or `false`.

File: core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java [hunk 3]
```diff
@@ -893,4 +894,31 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
...
+            res += (", scripted_upsert[" + scriptedUpsert + "]");
+        }
+        if (detectNoop) {
```
Comment (DaveCTurner) [inline core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java]: This is `boolean` - I think it'd be better to include it in the output whether it's `true` or `false`.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
It looks like you rewrote history to remove the commit that Tim looked at earlier so any comments he made are now lost. It makes it easier on reviewers if you just add commits to your branch. We squash the whole branch when it's merged so don't worry about making a noisy log at this stage. Please could you avoid rewriting history in future?

File: core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java (hunk 1 above)
Comment (zzt93) [reply core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java]: Fixed. As far as I see, the readability is affected, and as far as I know, java compiler will replace it with StringBuilder automatically now.

File: core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java (hunk 2 above)
Comment (zzt93) [reply core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java]: Fixed

File: core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java (hunk 3 above)
Comment (zzt93) [reply core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java]: Fixed. One more field which is a boolean and not listed here is also updated.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by cbuescher** [comment]
@DaveCTurner looks like this is ready to pull in
@zzt93 would you rebase this PR to master, the "core" subproject was recently renamed to "server", this is the cause for the conflicts I think

**Comment by cbuescher** [comment]
@zzt93 something looks wrong with this branch after the last rebase, it looks like you merged in another branch? Please clean up the branch PR before we can merge this. Let me know if you need help.

**Comment by zzt93** [comment]
@cbuescher I just rebase my branch and push, but remote refused. So I pull and merge. What do you mean 'clean up'? Maybe to push force?

**Comment by cbuescher** [comment]
@zzt93 by "clean up" I mean that your branch should only contain the two commits that were actually reviewd (I think that would be e00e478 and 5e3ba8a). Whatever git workflow suits you is fine for that, for example you could just check out the current tip of master, cherry-pick those commits and then push that to your pr branch. Whatever works. I think the "merge" you mentioned might have messed things up.

**Comment by cbuescher** [comment]
Sorry, I mean 783a0e5 instead of 5e3ba8a.

**Comment by zzt93** [comment]
Do as you suggested now.

**Comment by DaveCTurner** [comment]
Sorry @zzt93, it looks like this has failed because #28071 (and the associated changes to CI) landed in the meantime. Could you merge `master` in again?

**Comment by zzt93** [comment]
The jenkins log says:

```
01:42:25 * What went wrong:
01:42:25 Execution failed for task ':plugins:discovery-ec2:thirdPartyAudit'.
01:42:25 > CLASSES ARE MISSING! [javax/xml/bind/JAXBContext.class]
```

@DaveCTurner So what should I do?

**Comment by DaveCTurner** [comment]
I think this was fixed in aded32f48fc8dae8c67736e7c043a5aea665cfcd which was pushed after this PR, so something like `git pull upstream master && git push` should suffice.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by DaveCTurner** [comment]
Thanks for your contribution @zzt93, it's much appreciated. Merged to master as 1335232 and backported to 6.x as 5a82c93.

**Pull request merged**